tenacity/_asyncio.py

```diff
@@ -30,19 +30,31 @@
 WrappedFn = t.TypeVar("WrappedFn", bound=t.Callable[..., t.Awaitable[t.Any]])
 
 
-def asyncio_sleep(duration: float) -> t.Awaitable[None]:
+async def _portable_async_sleep(seconds: float) -> None:
+    # If trio is already imported, then importing it is cheap.
+    # If trio isn't already imported, then it's definitely not running, so we
+    # can skip further checks.
+    if "trio" in sys.modules:
+        # If trio is available, then sniffio is too
+        import trio
+        import sniffio
+
+        if sniffio.current_async_library() == "trio":
+            await trio.sleep(seconds)
+            return
+    # Otherwise, assume asyncio
     # Lazy import asyncio as it's expensive (responsible for 25-50% of total import overhead).
     import asyncio
 
-    return asyncio.sleep(duration)
+    await asyncio.sleep(seconds)
```

Comment:
The reason was is because that's how #295 did it and I didn't think about it when merging it with main :sweat_smile: 
Thinking about it now, doing it like #450 would be more performant (skipping `sniffie.current_async_library` check whenever calling sleep) - but it could maybe break if you're decorating an anyio test function that is parametrized against different async backends? Not 100% sure, but I can try to write a test that would trigger that.

Reply:
nowait I'm dumb, briefly forgot how async works. Yeah this makes no functional difference given the current code in `AsyncRetrying.__call__` where it immediately awaits the function after calling it. So it's just a question of whether one prefers an `async def` that itself calls `await` and returns `None` vs having a sync `def` that returns an `Awaitable[None]`. In general I find the former to be more readable and intuitive, but am fine with either.

Reply:
I'd prefer to not change things that don't need to be changed 😁

Reply:
fixed :)